DRODUtil/Util3_0.cpp

```diff
@@ -979,6 +980,11 @@ const
 		  //Add element.
 		  string strName = szName;
 		  UINT dwValue = convertToUINT(szValue);
+		  if (std::find(usedMids.begin(), usedMids.end(), dwValue) != usedMids.end()) {
```

Comment:
I like this addition as a sanity check.

Suggest using `set<UINT>` instead of `vector<UINT>`, as only one instance of each value is maintained in the data structure.
This reduces algorithmic cost for this loop from N^2 to N log N.

This check can then be simplified to `if (usedMids.count(dwValue) != 0) { ...`
(which has log N instead of N search time).

Reply:
I tried but it threw compilation errors I couldn't understand - I'll try again.